trieste/acquisition/function/function.py

```diff
@@ -1034,6 +1035,608 @@ def __call__(self, x: TensorType) -> TensorType:
         return tf.reduce_mean(batch_improvement, axis=-1, keepdims=True)  # [..., 1]
 
 
+class BatchExpectedImprovement(SingleModelAcquisitionBuilder[ProbabilisticModel]):
+    """Accurate approximation of the batch expected improvement, using the
+    method of Chvallier and Ginsbourger :cite:`chevalier2013fast`.
+
+    Internally, this uses a highly accurate approximation of the cumulative
+    density function of the multivariate Gaussian, developed by Alan Genz
+    :cite:`genz2016numerical`.
+    """
+
+    def __init__(
+        self,
+        sample_size: int,
+        *,
+        jitter: float = DEFAULTS.JITTER,
+    ):
+        """Initialise the BatchExpectedImprovement instance.
+
+        :param sample_size: int, number of Sobol samples to use.
+        :param jitter: float, amount of jitter for Cholesky factorisations.
+        """
+
+        tf.debugging.assert_positive(sample_size)
+        tf.debugging.assert_greater_equal(jitter, 0.0)
+
+        self._sample_size = sample_size
+        self._jitter = jitter
+
+    def __repr__(self) -> str:
+        """"""
+
+        return f"BatchExpectedImprovement({self._sample_size!r}, " f"jitter={self._jitter!r})"
+
+    def prepare_acquisition_function(
```

Comment:
it makes me sad that this is exactly the EI code or the MCEI one, except the call to `batch_expected_improvement` rather than `expected_improvement`. 

Is this OK? I can think of several solutions but all would look like hacky researcher code so it might be preferable to have some code duplication? 

 @uri-granta (possibly there's a good refactoring solution that may well be done in another OR)

Reply:
Yes, that's a good point. I don't know what's best here -- perhaps the `BatchMonteCarloExpectedImprovement` can handle both, with an option to use Ginsbourger if we want it to?

However, I think that it might be a little cleaner to keep these acquisition functions separate, as they might have quite different properties.

Reply:
I'd recommend checking it in like this and we can consider refactoring later (as we have done a few times now for other things).